src/fs/filter.rs

```diff
@@ -240,7 +242,19 @@ impl SortField {
             Self::Name(ABCabc)  => natord::compare(&a.name, &b.name),
             Self::Name(AaBbCc)  => natord::compare_ignore_case(&a.name, &b.name),
 
-            Self::Size          => a.metadata.len().cmp(&b.metadata.len()),
+            Self::Size => {
+                let recursive_map = RECURSIVE_SIZE_HASHMAP.lock().unwrap();
```

Comment:
I don't like that we have to use the hashmap and the lock for the non-recursive case.